engine/table/src/main/java/io/deephaven/engine/table/impl/hierarchical/TreeTableFilter.java

```diff
@@ -106,12 +111,6 @@ public MemoizedOperationKey getMemoKey() {
      */
     private final ColumnSource parentIdSource;
 
-    /**
```

Comment:
Remind me what magic you did to make the SwapListener not necessary for referential integrity.

Reply:
The SwapListener is held during initialization, but disappears from the DAG in its `end` method on success, by swapping the referent of the parent's `SimpleReference` to the listener from itself to the eventual listener.

Specifically, L145-147:
```
            referenceForSource.swapDelegate(initialDelegate, eventualListener instanceof LegacyListenerAdapter
                    ? (LegacyListenerAdapter) eventualListener
                    : new WeakSimpleReference<>(eventualListener));
```